chain/chunks/src/lib.rs

```diff
@@ -995,251 +994,217 @@ impl ShardsManager {
         let (started, key, response) = self.prepare_partial_encoded_chunk_response(request);
 
         let elapsed = started.elapsed().as_secs_f64();
-        let labels = [key, if response.is_some() { "ok" } else { "failed" }];
+        let labels = [
+            key,
+            if response.parts.is_empty() && response.receipts.is_empty() { "failed" } else { "ok" },
+        ];
         metrics::PARTIAL_ENCODED_CHUNK_REQUEST_PROCESSING_TIME
             .with_label_values(&labels)
             .observe(elapsed);
 
-        if let Some(response) = response {
-            self.peer_manager_adapter.do_send(
-                PeerManagerMessageRequest::NetworkRequests(
-                    NetworkRequests::PartialEncodedChunkResponse { route_back, response },
-                )
-                .with_span_context(),
-            );
-            return true;
-        }
-        false
+        self.peer_manager_adapter.do_send(
+            PeerManagerMessageRequest::NetworkRequests(
+                NetworkRequests::PartialEncodedChunkResponse { route_back, response },
+            )
+            .with_span_context(),
+        );
     }
 
     fn prepare_partial_encoded_chunk_response(
         &mut self,
         request: PartialEncodedChunkRequestMsg,
-    ) -> (std::time::Instant, &'static str, Option<PartialEncodedChunkResponseMsg>) {
+    ) -> (std::time::Instant, &'static str, PartialEncodedChunkResponseMsg) {
         // Try getting data from in-memory cache.
         let started = Instant::now();
-        if let Some(entry) = self.encoded_chunks.get(&request.chunk_hash) {
-            let response = Self::prepare_partial_encoded_chunk_response_from_cache(request, entry);
+        let PartialEncodedChunkRequestMsg { chunk_hash, part_ords, mut tracking_shards } = request;
+        let mut response = PartialEncodedChunkResponseMsg {
+            chunk_hash: chunk_hash.clone(),
+            parts: vec![],
+            receipts: vec![],
+        };
+        let mut part_ords = part_ords.into_iter().collect::<HashSet<_>>();
+        if part_ords.is_empty() && tracking_shards.is_empty() {
+            return (started, "none", response);
```

Comment:
I think it's not good code practice to have the key returned in a string format, previously the string was sorta of ok since it is only used for debug purposes. Let's create an enum now that it actually matters for the key code logic.

Reply:
Yep!